### PR TITLE
Add deepsource code coverage

### PR DESCRIPTION
**Pull request by matmair**
This PR is a follow-up tp #5714 and adds code coverage reporting to deepsource (the coverage used for coveralls is reused)

### CODE REVIEW

**Comment by matmair** [comment]
The function of this depends on a secret being set, that was transferred on another medium

**Comment by matmair** [comment]
@SchrodingersGat this would be ready for merge

**Comment by wolflu05** [comment]
What is the improvement? Does this mean we can drop coveralls?

**Comment by matmair** [comment]
We have a full view of the problem in the SAST solution. I would not drop coveralls, we have years of history there.

**Comment by wolflu05** [comment]
Makes sense.

**Comment by wolflu05** [comment]
You can also commit without changing something with the --allow-empty argument

**Comment by matmair** [comment]
Yes but that wouldn't trigger a server CI run.

**Comment by SchrodingersGat** [comment]
Not sure why this failed - `DEEPSOURCE_DSN` is set

**Comment by wolflu05** [comment]
Secrets are not used for pull_request triggers for security reasons. Otherwise anyone could just read out every secret by sending them to their servers via curl. @matmair needs to set this secret in his repo and run the actions on his repo.

**Comment by SchrodingersGat** [comment]
> Secrets are not used for pull_request triggers from forks for security reasons. Otherwise anyone could just read out every secret by sending them to their servers via curl when modifying the workflow. @matmair needs to set this secret in his repo and run the actions on his repo.

Ah, of course!

File: .github/workflows/qc_checks.yaml [hunk 1]
```diff
@@ -217,6 +217,13 @@ jobs:
         uses: coverallsapp/github-action@v2
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload Coverage to Deepsource
```
Comment (SchrodingersGat) [inline .github/workflows/qc_checks.yaml:220]: You will need to adjust this action to only run on push, not PR

**Comment by wolflu05** [comment]
But if the action relies on this secret, I guess we need to search for a different approach? How does the coveralls auth works?

**Comment by SchrodingersGat** [comment]
I think coveralls relies on a github integration which you can authorize - rather than a repo secret

**Comment by matmair** [comment]
I have added the DSN to my repo and it worked in my out-of-tree repos but I will just disable it on PRs

**Comment by matmair** [comment]
I think we have a differing understanding of the goal here: Mine is to add code coverage indicators to the SAST tool. This is not meant as a replacement for coveralls.

**Comment by wolflu05** [comment]
Oh ok, makes sense now.

**Comment by SchrodingersGat** [comment]
@matmair is this OK to merge in now, or needs further work?

**Comment by matmair** [comment]
@SchrodingersGat this is ok to merge

**Pull request merged**